### DISCUSSION TITLE
[Solved]Error: No id found

### DISCUSSION THREAD

**Discussion by rl-devself** [django-commons/django-unicorn#119] [Q&A] [answered]
Hello, 

I learned Django a few mounths ago, and I was looking for a front-end framework when I found out about django-unicorn. It's a very interesting project, I appreciate your work. 

However, I'm having the same error with all django-unicorn examples. Whenever I do some action on the component, it disappear, and I have this error in the browser console : Error: No id found. 

Even thought I copied exactly the same code for the todo example.
Should I specify an id when including the component in my django html template? 

Thank you for any help,

Devself

**Comment by rl-devself** [comment, marked as answer]
I started again and it works fine :)

**Reply by adamghill** [reply]
That's great to hear that it works now! If you run into other issues (or if the documentation could be improved to avoid a similar issue for others), please let me know. 👍

**Comment by chinoosniel** [comment]
Only work for my if i use PYCHARM else i have the same problem, what do you do for get it work?

**Reply by adamghill** [reply]
Can you create a GitHub repo that demonstrates the problem? That way I can see the code you are trying and help troubleshoot. Thanks!

**Comment by chenatlas** [comment]
I had this exact error happen as well, I didn't know why, copied the examples from the tutorial
after removing `<!-- unicorn/templates/unicorn/todo.html -->` as the first line (copied that as well), it started working.
I hope that helps!

**Reply by trolldbois** [reply]
Confirmed, if one puts anything before the first <div> in the component template, it breaks.
No comment, no tag, no text allow before div.

**Reply by adamghill** [reply]
I think I have a fix for this going into master right now: https://github.com/adamghill/django-unicorn/commit/b6d5604770ec8f4c080cac3430da8a13b6e72eaa. I'll let you know once it's released.

**Reply by adamghill** [reply]
This should now be fixed in 0.33.0: https://github.com/adamghill/django-unicorn/releases/tag/0.33.0. Feel free to make an issue if you still the problem after upgrading.

**Reply by trolldbois** [reply]
Confirmed as Fixed

**Reply by OBKF** [reply]
Comments breaks the component, thanks for sharing the solution.
